packages/terraform-modules-public/aws/testnet/modules/vpc/main.tf

```diff
@@ -0,0 +1,246 @@
+resource "aws_vpc" "celo" {
+    cidr_block = var.cidr_blocks.vpc
+
+    tags = {
+        Name = var.name
+    }
+}
+
+resource "aws_default_security_group" "default" {
+    vpc_id = aws_vpc.celo.id
+
+    ingress {
+        protocol    = -1
+        self        = true
+        from_port   = 0
+        to_port     = 0
+    }
+}
+
+resource "aws_internet_gateway" "igw" {
+    vpc_id = aws_vpc.celo.id
+
+    tags = {
+        Name = "celo-internet-gateway"
+    }
+}
+
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+
+module "celo_public_subnet_az1" {
+    source = "./modules/subnet-public"
+
+    vpc_id                          = aws_vpc.celo.id
+    cidr_block                      = var.cidr_blocks.subnet_az1_public
+    internet_gateway_id             = aws_internet_gateway.igw.id
+    availability_zone_id            = data.aws_availability_zones.available.zone_ids[0]
+    allowed_ssh_clients_cidr_block  = var.cidr_blocks.allowed_ssh_clients
+}
+
+module "celo_private_subnet_az1" {
+    source = "./modules/subnet-private"
+
+    vpc_id                          = aws_vpc.celo.id
+    cidr_block                      = var.cidr_blocks.subnet_az1_private
+    availability_zone_id            = data.aws_availability_zones.available.zone_ids[0]
+    nat_gateway_id                  = module.celo_public_subnet_az1.nat_gateway_id
+    vpc_cidr_block                  = aws_vpc.celo.cidr_block
+}
+
+module "celo_public_subnet_az2" {
+    source = "./modules/subnet-public"
+
+    vpc_id                          = aws_vpc.celo.id
+    cidr_block                      = var.cidr_blocks.subnet_az2_public
+    internet_gateway_id             = aws_internet_gateway.igw.id
+    availability_zone_id            = data.aws_availability_zones.available.zone_ids[1]
+    allowed_ssh_clients_cidr_block  = var.cidr_blocks.allowed_ssh_clients
+}
+
+module "celo_private_subnet_az2" {
+    source = "./modules/subnet-private"
+
+    vpc_id                          = aws_vpc.celo.id
+    cidr_block                      = var.cidr_blocks.subnet_az2_private
+    availability_zone_id            = data.aws_availability_zones.available.zone_ids[1]
+    nat_gateway_id                  = module.celo_public_subnet_az2.nat_gateway_id
+    vpc_cidr_block                  = aws_vpc.celo.cidr_block
+}
+
+resource "aws_security_group" "attestation_service" {
+    name            = "celo-attestation-service"
+    vpc_id          = aws_vpc.celo.id
+
+    ingress {
```

Comment:
Not a big deal but is there any functionality issue for using inline rules for some security groups and `aws_security_group_rule` for others? I used to use `aws_security_group_rule` because inline rules used to cause some issues with create&destroy the SG which caused some downtime, but I have read it was fixed some time ago in the aws provider.

Reply:
Inline rules have worked well for me when I change them. In my testing, terraform has properly done SG updates instead of create/destroy.

Reply:
Nice.